features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptOrderByItemTokenGenerator.java

```diff
@@ -101,7 +101,9 @@ private Collection<OrderByItem> getOrderByItems(final SQLStatementContext sqlSta
         }
         Collection<OrderByItem> result = new LinkedList<>();
         SelectStatementContext statementContext = (SelectStatementContext) sqlStatementContext;
-        result.addAll(statementContext.getOrderByContext().getItems());
+        if (!statementContext.getOrderByContext().isGenerated()) {
```

Comment:
Why we remove generated?

Reply:
To address issue #4565, the sharding feature automatically adds the ORDER BY clause when processing the "SELECT distinct amount FROM t_account" statement. However, for the account_id column that has encryption rules configured, when processing the same statement, it only needs to rewrite the encrypted column name without adding the ORDER BY clause.

The presence of the orderByContext member variable in the SelectStatementContext class causes the problem described in issue #20096 when rewriting the SQL. It mistakenly assumes that the original SQL is "SELECT distinct amount FROM t_account ORDER BY amount".

References:
https://github.com/apache/shardingsphere/blob/5.3.2/test/it/rewriter/src/test/resources/scenario/sharding/case/dml/select.xml#L101
https://github.com/apache/shardingsphere/pull/4597